platform/osx/display_server_osx.mm

```diff
@@ -3803,9 +3803,11 @@ Point2i window_position(
 	}
 
 	//destroy all windows
-	for (Map<WindowID, WindowData>::Element *E = windows.front(); E; E = E->next()) {
-		[E->get().window_object setContentView:nil];
-		[E->get().window_object close];
+	for (Map<WindowID, WindowData>::Element *E = windows.front(); E;) {
+		Map<WindowID, WindowData>::Element *F = E;
+		E = E->next();
+		[F->get().window_object setContentView:nil];
+		[F->get().window_object close];
```

Comment:
That looks weird, what was the problem here? Did `[E->get().window_object close];` somewhat invalid `E` so `E->next()` as increment would no longer work?

Reply:
`window_object close` will call `windowWillClose` and erase element from the map.